src/torch_models.py

```diff
@@ -146,6 +146,105 @@ def _fit(self, inputs: Array, outputs: Array) -> None:
         logging.info(f"Loaded best model with loss: {loss:.5f}")
 
 
+class ImplicitMLPRegressor(Regressor):
+    """A regressor implemented via an energy function (binary classifier).
```

Comment:
wait, how is a binary classifier an energy function? An energy function is a mapping from the input to an unnormalized score.

Reply:
agreed this is wrong. discussed offline, going to merge this anyway and then try to do it correctly after (issue #672)

Reply:
fix the comment though